gnomad_constraint/utils/constraint_basics.py

```diff
@@ -81,3 +93,64 @@ def prepare_ht_for_constraint_calculations(ht: hl.Table):
     # Filter out locus with undefined exome coverage
     ht = ht.filter(hl.is_defined(ht.exome_coverage))
     return ht
+
+
+def create_constraint_training_dataset(
+    exome_ht: hl.Table,
+    context_ht: hl.Table,
+    mutation_ht: hl.Table,
+    max_af: float = 0.001,
+    kept_annotations: Tuple[str] = (
+        "context",
+        "ref",
+        "alt",
+        "methylation_level",
+        "exome_coverage",
+    ),
+    pops: Tuple[str] = POPS,
+    grouping: Tuple[str] = ("exome_coverage",),
+    partition_hint: int = 100,
+) -> hl.Table:
+    """
+    Count the observed variants and possible variants by exome coverage.
+
+    :param exome_ht: Preprocessed exome Table.
+    :param context_ht: Preprocessed context Table.
+    :param mutation_ht: Preprocessed mutation rate Table.
+    :param max_af: Variants with AF above than AF cutoff will be removed, defaults to 0.001.
+    :param kept_annotations: Annotations to keep in the context Table.
+    :param pops: List of populations. Defaults to `POPS`.
+    :param grouping: Annotations other than 'context', 'ref', 'alt' to group by when counting variants, defaults to 'exome_coverage'.
+    :param partition_hint: Target number of partitions for aggregation. Defaults to 100.
+    :return: Table with observed variant and possible variant count.
+    """
+    freq_expr = exome_ht.freq[0]
+
+    keep_criteria = (
+        (freq_expr.AC > 0) & exome_ht.pass_filters & (freq_expr.AF <= max_af)
+    )
+
+    ht = count_variants(
+        filter_vep_transcript_csqs(exome_ht.filter(keep_criteria)).select(
+            *list(kept_annotations) + ["freq"]
+        ),
+        additional_grouping=grouping,
+        partition_hint=partition_hint,
+        count_downsamplings=pops,
+        use_table_group_by=True,
+        max_af=max_af if max_af else None,
```

Comment:
We don't want to use this anymore since we are doing the AF filter above instead
```suggestion
```